coredocument/read_acls.go

```diff
@@ -54,3 +58,59 @@ func appendRole(roles []*coredocumentpb.RoleEntry, role *coredocumentpb.Role) []
 		Role:    role,
 	})
 }
+
+// ReadAccessValidator defines validator functions for peer.
+type ReadAccessValidator interface {
```

Comment:
Don't we want to use similar structure as https://github.com/centrifuge/go-centrifuge/blob/develop/p2p/receiver/validator.go#L14 for example? Or it is too much overhead?

Reply:
We thought about it but its just overhead and this will evolve as we add access token and nft access